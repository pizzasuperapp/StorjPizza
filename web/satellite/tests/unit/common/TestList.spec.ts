// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { mount } from '@vue/test-utils';

import TestList from '../mock/components/TestList.vue';

describe('TestList.vue', () => {
    it('should render list of primitive types', function () {
        const wrapper = mount(TestList, {
            propsData: {
                onItemClick: jest.fn(),
            },
        });
        expect(wrapper).toMatchSnapshot();
    });

    it('should retrieve callback', function () {
        const onPressSpy = jest.fn();

        const wrapper = mount(TestList, {
            propsData: {
                onItemClick: onPressSpy,
            },
        });
        wrapper.find('.item-component__item').trigger('click');
        expect(onPressSpy).toHaveBeenCalledTimes(1);
    });
});
