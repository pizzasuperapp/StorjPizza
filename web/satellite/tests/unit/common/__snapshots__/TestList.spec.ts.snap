// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestList.vue should render list of primitive types 1`] = `
<div class="item-component">
  <h1 class="title item-component__item">1</h1>
  <h1 class="title item-component__item">2</h1>
  <h1 class="title item-component__item">3</h1>
</div>
`;
